helpers/clean_pnp_backdata.py

```diff
@@ -0,0 +1,320 @@
+import pandas as pd
+import argparse
+import toml
+
+
+def convert_column_datatypes(df):
+    """ Function to convert the column datatypes of a dataframe if they appear
+    within backdata_schema.toml.
+
+    Args:
+        df (pd.DataFrame): The dataframe to convert the column datatypes.
+    Return:
+        DF (pd.DataFrame): The dataframe with the converted column datatypes.
+
+    """
+    with open('./config/backdata_schema.toml', 'r') as f:
+        datatypes_config = toml.load(f)
+
+    datatypes_dict = {}
+
+    for column, datatype in datatypes_config.items():
+        datatypes_dict.update({column: datatype["Deduced_Data_Type"]})
+
+    for i in list(df.columns):
+        if i in list(datatypes_dict.keys()):
+            try:
+                df[i] = df[i].astype(datatypes_dict[i])
+            except:
+                pass
+        else:
+            continue
+
+    return df
+
+
+def get_status_encoded(df):
+    """ Function to populate the statusencoded column based on the mapping of the
+    'map cora status to column "status"' column.
+
+    Args:
+        df (pd.DataFrame): The dataframe to populate the statusencoded column.
+    Return:
+        DF (pd.DataFrame): The dataframe with the populated statusencoded column.
+    """
+    cora_to_status_encoded_dict = {200: 100,
+                                   100: 101,
+                                   1000: 102,
+                                   400: 200,
+                                   500: 201,
+                                   600: 210,
+                                   800: 211,
+                                   1200: 302,
+                                   1300: 303,
+                                   900: 304,
+                                   1400: 309}
+
+    df["statusencoded"] = df['map cora status to column "status"'].map(
+        cora_to_status_encoded_dict
+    )
+
+    return df
+
+
+def identify_key_business(df):
+    """ Function to identify the key business using reference column & key
+    busineses lookup table.
+
+    Args:
+        df (pd.DataFrame): The dataframe to identify the key business columns.
+    Return:
+        df (pd.DataFrame): The dataframe with the identified key business columns.
+    """
+    # get key businesses
+    key_businesses_df = pd.read_csv('R:/KEYS 2023.csv')

```

Comment:
Hi Ryan, this probably isn't a good place to store "our" data, a large business area owns the sharedrive OF4 that we call "R". So it might be better to put the files we're working with to an area within our bit, which is "BERD results system development 2023". I suggest alongside the other 2021 pnp data.

Reply:
OHHH my bad, I only put the file here as temporary measure, and then forgot about it. I will move this file.